### ISSUE TITLE
Follow up on grpc interceptor fix

### ISSUE THREAD

**Issue by eaddingtonwhite** [momentohq/client-sdk-python#149]
See the following PR for more context. 
https://github.com/momentohq/client-sdk-python/pull/148

TLDR: We are using grpc interceptors to inject auth token and user agent metadata per request. When client is using `ddtrace` library its passing along metadata as `list` instead of `grpc.aio.Metadata` object which was causing sdk to choke. 

This ticket was meant to be a follow-up for the initial fix we put in for this to make our interceptor more defensive in how it handled inbound client_request_metadata.

Longer term I would like to take the following actions:
1. I would like to revisit why we are even using client interceptors to inject this metadata into requests in the first place. We are the ones initiating requests with GRPC so we have the opportunity to inject all the metadata we need to make request upfront as we build the request object. We already are injecting cache-name dynamically on request metadata this way from with in our SDK data client [here for example](https://github.com/momentohq/client-sdk-python/blob/main/src/momento/aio/_scs_data_client.py#L149) using [this](https://github.com/momentohq/client-sdk-python/blob/50a69563683fd8e37cfe9707121029192f3a5f21/src/momento/_utilities/_data_validation.py#L10-L11) utility method. We could just make this bit more flexible to set all metadata upfront and thus eliminate the need for us to even have yet another interceptor in the mix on request lifecycle. I think by doing all in side our clients as we build up requests will be easier to debug and maintain.
2. Work with Datadog longer term and try and patch their interceptor to play better with others and follow proper typing in gRPC python library.
